elementary/clients/api/api_client.py

```diff
@@ -1,14 +1,15 @@
 from collections import defaultdict
-from typing import Any
+from typing import Any, Union
 
 from elementary.clients.dbt.dbt_runner import DbtRunner
+from elementary.clients.dbt.slim_dbt_runner import SlimDbtRunner
 from elementary.utils.log import get_logger
 
 logger = get_logger(__name__)
 
 
 class APIClient:
-    def __init__(self, dbt_runner: DbtRunner):
+    def __init__(self, dbt_runner: Union[DbtRunner, SlimDbtRunner]):
```

Comment:
Why not change the type to be simply `BaseDbtRunner`?

Reply:
IMO BaseDbtRunner is too generic.
Here I know exactly which DbtRunners are supported, but it is not that matters currently because we support all :)

Reply:
I think that this class can actually support any class that inherits from BaseDbtRunner?
E.g. if I want to initialize it with ItamarCustomDbtRunner that implements the same methods, it should work, no?

I think it makes sense to specify the minimal interface that is required, that is the point of inheritance :)